ambari-server/src/main/java/org/apache/ambari/server/events/publishers/AgentCommandsPublisher.java

```diff
@@ -111,7 +111,10 @@ public void sendAgentCommand(Multimap<Long, AgentCommand> agentCommands) throws
                 if (!clusterDesiredConfigs.containsKey(clusterId)) {
                   clusterDesiredConfigs.put(clusterId, clusters.getCluster(clusterId).getDesiredConfigs());
                 }
-              } catch (NumberFormatException|AmbariException ignored) {}
+              } catch (NumberFormatException|AmbariException e) {
+                LOG.error("Exception on sendAgentCommand", e);
+                throw new RuntimeException(e);
```

Comment:
this should be incomtablie change right as it was ignore before. and actionschduler throws only ambariexception.

Reply:
Could I change `RuntimeException` to `AmbariException` ?
If I don't rethrow the exception, it hangs on `Test Kerberos Client` or `Distribute Keytabs` and you would wait until it gets timeout.
![image](https://user-images.githubusercontent.com/12639125/187306260-316615b7-b558-4ef3-b4c9-07916d39d09e.png)
![image](https://user-images.githubusercontent.com/12639125/187306270-a2412d0d-f5bb-4341-bbeb-c3effa291199.png)

Reply:
Let's through AmbariException.

Reply:
I think `RuntimeException` or `AmbariRuntimeException` will be handled on `ActionScheduler#run`. (https://github.com/apache/ambari/blob/branch-2.7/ambari-server/src/main/java/org/apache/ambari/server/actionmanager/ActionScheduler.java#L352-L354)

So, I just changed `RuntimeException` to `AmbariRuntimeException`.
If you want to use `AmbariException`, many changes are needed.
Isn't it the right way to use `AmbariRuntimeException` instaed of `AmbariException`?

Reply:
There is also another example line which use `AmbariRuntimeException` when a step fails in the middle of a stage.
=> https://github.com/apache/ambari/blob/branch-2.7/ambari-server/src/main/java/org/apache/ambari/server/events/publishers/AgentCommandsPublisher.java#L178

Reply:
ok.. I also feel, AmbariRuntimeException should be fine.